torch/_prims/wrappers.py

```diff
@@ -194,3 +193,51 @@ def _fn(*args, out=None, **kwargs):
     _fn.__annotations__ = fn.__annotations__
     _fn.__annotations__["out"] = TensorLikeType
     return _fn
+
+
+def out_wrapper_multi(*out_names):
+    def go(fn: Callable) -> Callable:
+        @wraps(fn)
+        def _fn(*args, **kwargs):
+            out_kwargs = {}
+            has_out_kwargs = None
+            for o in out_names:
+                out_kwargs[o] = kwargs.pop(o, None)
+                # Either all of the out kwargs are set or none of them
+                if has_out_kwargs is None:
+                    has_out_kwargs = out_kwargs[o] is not None
+                else:
+                    assert has_out_kwargs == (out_kwargs[o] is not None)
+            result = fn(*args, **kwargs)
+            assert isinstance(result, tuple)
+            if has_out_kwargs:
+                final_result = []
+                for i, o in enumerate(out_names):
+                    out = out_kwargs[o]
+                    assert isinstance(out, TensorLike)
+                    out = _maybe_resize_out(out, result[i].shape)
+                    final_result.append(_safe_copy_out(copy_from=result[i], copy_to=out))  # type: ignore[arg-type]
+                return tuple(final_result)
+            return result
+
+        sig = inspect.signature(fn)
+        out_params = []
+        for o in out_names:
+            out_params.append(
+                inspect.Parameter(
+                    o,
+                    kind=inspect.Parameter.KEYWORD_ONLY,
+                    default=None,
+                    annotation=TensorLikeType,
+                )
+            )
+        params = chain(sig.parameters.values(), out_params)
+        _fn.__signature__ = inspect.Signature(  # type: ignore[attr-defined]
+            parameters=params, return_annotation=sig.return_annotation  # type: ignore[arg-type]
+        )
+        _fn.__annotations__ = fn.__annotations__
+        for o in out_names:
+            _fn.__annotations__[o] = TensorLikeType
+        return _fn
```

Comment:
Neat -- in the future should the regular out_wrapper be replaced or made a sugar for this?